src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -6911,6 +6911,9 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="ERR_LambdaWithAttributesToExpressionTree" xml:space="preserve">
     <value>A lambda expression with attributes cannot be converted to an expression tree</value>
   </data>
+  <data name="ERR_RecordStructConstructorCallsDefaultConstructor" xml:space="preserve">
+    <value>A 'this' initializer for a 'record struct' constructor must call the primary constructor or an explicitly declared constructor.</value>
```

Comment:
>A 'this' initializer for a 'record struct' constructor must call the primary constructor or an explicitly declared constructor. [](http://example.com/codeflow?start=11&length=127)

The wording looks too general. The requirement is only for records with primary constructors, I think the message should reflect that. See, for example, how we do that for ```UnexpectedOrMissingConstructorInitializerInRecord```:

```
  <data name="ERR_UnexpectedOrMissingConstructorInitializerInRecord" xml:space="preserve">
    <value>A constructor declared in a record with parameter list must have 'this' constructor initializer.</value>
  </data>
```